src/proto/flwr/proto/recordset.proto

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ==============================================================================
+
+syntax = "proto3";
+
+package flwr.proto;
+
+import "flwr/proto/task.proto";
+
+message Array {
+  string dtype = 1;
+  repeated int32 shape = 2;
+  string stype = 3;
+  bytes data = 4;
+}
+
+message ParametersRecord {
+  repeated string data_keys = 1;
+  repeated Array data_values = 2;
+}
+
+message MetricsRecord { map<string, Value> data = 1; }
+
+message ConfigsRecord { map<string, Value> data = 1; }
```

Comment:
`ConfigsRecord` doesn't allow `bool` yet. That's sth we should fix in a separate PR by adding it to `ConfigsRecord`.

The more general question: should we define ProtoBuf messages that follow the same naming scheme as their Python dataclass counterpart?

Reply:
The reason for reusing `Value` instead of creating a new message `ConfigsRecordValue` is the same as above (`MetricsRecord`).

Re: naming. IMO, we should use different names or even store them in a different way in a TaskIns. There's no benefits to keep them strictly equivalent. I think the ProtoBuf messages is solely to transfer contents over the wire, which is different from the purpose of introducing record types. And hence I think we don't need to follow the same naming scheme and even don't need to have a counterpart for each dataclass record.

Reply:
I agree that we do not need to aim for a strict 1:1 mapping from Python to ProtoBuf.

I do think however that there is an advantage to keeping them close. As stated in the other comment, we want to implement Flower clients in languages other than Python. This usually starts with compiling existing ProtoBuf messages. If those messages are close to the Python level, it will be easier for others to implement the Java/C++/... counterpart.

In addition to that, it will maintenance of the Python client easier as well. In the case of `MetricsRecord`, for example, we would not need to check for unsupported `Value` types if we have a `MetricsRecord` on the ProtoBuf level that supports the exact set of types that the Python `MetricsRecord` supports.